### PR TITLE
tools/openocd: export configuration variables only to required targets

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is similar to #13452 but is applied to `OPENOCD_RESET_USE_CONNECT_ASSERT_SRST` and `OPENOCD_ADAPTER_INIT`.
Static checks are also added.

### Testing procedure

- Verify that affected boards (samr21-xpro, nucleos, b-l072z-lrwan1, frdm-kw41z, iotlab-m3, nrf52dk with PROGRAMMER=openocd, mulle) can still be flashed when the CPU in "usual" conditions but also in deep sleep mode.

### Issues/PRs references

Related to #10850

### CODE REVIEW

**Comment by kfessel** [comment]
There where some recent additions to openocd.sh which are problematic please see #13470

File: makefiles/tools/openocd.inc.mk [hunk 1]
```diff
@@ -18,3 +18,10 @@ OPENOCD_CONFIG ?= $(BOARDSDIR)/$(BOARD)/dist/openocd.cfg
 # Export OPENOCD_CONFIG to required targets
 OPENOCD_TARGETS = debug% flash% reset
 $(call target-export-variables,$(OPENOCD_TARGETS),OPENOCD_CONFIG)
+
+# Export OPENOCD_ADAPTER_INIT to required targets
+$(call target-export-variables,$(OPENOCD_TARGETS),OPENOCD_ADAPTER_INIT)
+
+# Export OPENOCD_RESET_USE_CONNECT_ASSERT_SRST to required targets
+OPENOCD_TARGETS = flash reset
+$(call target-export-variables,$(OPENOCD_TARGETS),OPENOCD_RESET_USE_CONNECT_ASSERT_SRST)
```
Comment (fjmolinas) [inline makefiles/tools/openocd.inc.mk]: No need for a variable here and it overwrites `OPENOCD_TARGETS = debug% flash% reset`

```suggestion
# Export OPENOCD_RESET_USE_CONNECT_ASSERT_SRST to required targets
$(call target-export-variables,flash% reset,OPENOCD_RESET_USE_CONNECT_ASSERT_SRST)
```

Comment (fjmolinas) [reply makefiles/tools/openocd.inc.mk]: Note that it should also be `flash% reset` to include `flash-only`

**Comment by aabadie** [comment]
Comment addressed @fjmolinas.

**Pull request merged**